src/openvic-simulation/economy/production/ResourceGatheringOperation.cpp

```diff
@@ -14,8 +24,228 @@ ResourceGatheringOperation::ResourceGatheringOperation(
 	output_quantity_yesterday { new_output_quantity_yesterday },
 	unsold_quantity_yesterday { new_unsold_quantity_yesterday },
 	size_multiplier { new_size_multiplier },
-	employees { std::move(new_employees) } {}
+	employees { std::move(new_employees) },
+	total_employees_count_cache {},
+	total_paid_employees_count_cache { 0},
+	total_owner_income_cache {},
+	total_employee_income_cache {}
+{}
 
-ResourceGatheringOperation::ResourceGatheringOperation(
-	ProductionType const& new_production_type, fixed_point_t new_size_multiplier
-) : ResourceGatheringOperation { new_production_type, new_size_multiplier, 0, 0, 0, {} } {}
+ResourceGatheringOperation::ResourceGatheringOperation() : ResourceGatheringOperation {
+	nullptr, fixed_point_t::_0(),
+	 fixed_point_t::_0(), fixed_point_t::_0()
+	 , fixed_point_t::_0(), {}
+} {}
+
+void ResourceGatheringOperation::initialise_for_new_game(ProvinceInstance& location) {
+	if(this->production_type == nullptr) {
+		output_quantity_yesterday = 0;
+		revenue_yesterday = 0;
+		return;
+	}
+
+	ProductionType const& production_type = *(this->production_type);
```

Comment:
Best not to create another variable with the same name as the member variable `production_type` as it makes the code ambiguous, either give it a different name or just use the pointer directly since you already know it's not null. Same throughout the rest of this file.